capstone-frontend/src/components/ClubPage.jsx

```diff
@@ -1,23 +1,72 @@
 import React, { Component } from 'react';
 
+import BookSearchTile from './BookSearchTile';
+import AssignmentCard from './AssignmentCard';
+ 
+import '../styles/Groups.css';
+
 
 class ClubPage extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      club: {}
+      club: {},
+      book: {},
+      assignments: [],
+      owner: {}, 
+      members: [], 
+      bookLists: []
     }
   }
 
-  fetchClubData = () => {
-    fetch(`/api/clubs?id=${this.props.match.params.id}`)
-        .then(response => response.json()).then(res => this.setState({club: res[0]}))
+  fetchData = async () => {
+    await fetch(`/api/clubs?id=${this.props.id}`)
+        .then(response => response.json()).then(clubJson => this.setState({club: clubJson[0]}))
         .catch(function(err) {
             //TODO #61: Centralize error output
             alert(err);
         });
-  }
 
+    await fetch(`/api/search?gbookId=${this.state.club.gbookID}`)
+        .then(response => response.json()).then(bookJson => this.setState({book: bookJson[0]}))
+        .catch(function(err) {
+            //TODO #61: Centralize error output
+            alert(err);
+        });
+
+    await fetch(`/api/assignments?clubID=${this.state.club.id}`)
+        .then(response => response.json()).then(assignmentJson => this.setState({assignments: assignmentJson}))
+        .catch(function(err) {
+            //TODO #61: Centralize error output
+            alert(err); 
+        });
+
+    await fetch(`/api/user?id=${this.state.club.ownerID}`)
+        .then(response => response.json()).then(ownerJson => this.setState({owner: ownerJson}))
+        .catch(function(err) {
+            //TODO #61: Centralize error output
+            alert(err); 
+        });
+
+    for (let i = 0; i < this.state.club.memberIDs.length; i++) {
```

Comment:
Just something to consider for the future... it might be worthwhile to update the UserServlet to accept a list of user IDs and replace this with a single `fetch`.

Reply:
I agree that's a good idea. I'll discuss this with Mauro and see if we can build out that functionality. Would also likely be useful for all servlets (ie. when fetching many books in a BookList, or all of a user's clubs, etc)